pages/app/index.tsx

```diff
@@ -80,6 +71,10 @@ function App() {
   );
 }
 
+// The VR class needs to be set before any React rendering takes occurs.
```

Comment:
Something very interesting happens when the `useVisualRefresh` hook gets called much earlier, i.e., during metrics. VR is only applied half-way through rendering and is causing inconsistent results. That's why the visual regression tests failed.

Curious how you feel about this change, @just-boris @rubencarvalho

Reply:
This is expected, because we do not follow our own visual refresh activation guide. Specifically, we run it inside `useEffect`, where the guideline has not to do so:

https://github.com/cloudscape-design/components/blob/07e7e371d5bb6560f5f0d24f4e7d1664fb26f856/pages/app/index.tsx#L59-L66

> The theme activates only on the first render. **Do NOT use React hooks** or lifecycle methods or JSX content for this class name, because it will not work correctly

Your addition makes sense, but some suggestions

1. Replace `document.getElementById('app')?` with `document.getElementById('app')!`, because if this code is missing we want to fail loud, not quiet 
2. Remove `useEffect` code, because it is not needed
3. Use hash history instead of manual parsing

```js
import { createHashHistory } from "history";

const history = createHashHistory();

// use history.location.search to parse the parameters, reusing with this code for consistency https://github.com/cloudscape-design/components/blob/07e7e371d5bb6560f5f0d24f4e7d1664fb26f856/pages/app/app-context.tsx#L40

<Router history={history}>...</Router>
```

Reply:
There is no such thing as automatic referesh, because the page reloads anyway:  https://github.com/cloudscape-design/components/blob/07e7e371d5bb6560f5f0d24f4e7d1664fb26f856/pages/app/index.tsx#L28-L30

This code could get some refactoring and simplification